javaparser-core/src/main/java/com/github/javaparser/ast/expr/AssignExpr.java

```diff
@@ -49,34 +54,56 @@
 
     public enum Operator implements Printable {
 
-        ASSIGN("=", null),
-        PLUS("+=", BinaryExpr.Operator.PLUS),
-        MINUS("-=", BinaryExpr.Operator.MINUS),
-        MULTIPLY("*=", BinaryExpr.Operator.MULTIPLY),
-        DIVIDE("/=", BinaryExpr.Operator.DIVIDE),
-        BINARY_AND("&=", BinaryExpr.Operator.BINARY_AND),
-        BINARY_OR("|=", BinaryExpr.Operator.BINARY_OR),
-        XOR("^=", BinaryExpr.Operator.XOR),
-        REMAINDER("%=", BinaryExpr.Operator.REMAINDER),
-        LEFT_SHIFT("<<=", BinaryExpr.Operator.LEFT_SHIFT),
-        SIGNED_RIGHT_SHIFT(">>=", BinaryExpr.Operator.SIGNED_RIGHT_SHIFT),
-        UNSIGNED_RIGHT_SHIFT(">>>=", BinaryExpr.Operator.UNSIGNED_RIGHT_SHIFT);
+        ASSIGN("="),
+        PLUS("+="),
+        MINUS("-="),
+        MULTIPLY("*="),
+        DIVIDE("/="),
+        BINARY_AND("&="),
+        BINARY_OR("|="),
+        XOR("^="),
+        REMAINDER("%="),
+        LEFT_SHIFT("<<="),
+        SIGNED_RIGHT_SHIFT(">>="),
+        UNSIGNED_RIGHT_SHIFT(">>>=");
 
         private final String codeRepresentation;
 
-        private final BinaryExpr.Operator binaryOperatorEquivalent;
-
-        Operator(String codeRepresentation, BinaryExpr.Operator binaryOperatorEquivalent) {
+        Operator(String codeRepresentation) {
             this.codeRepresentation = codeRepresentation;
-            this.binaryOperatorEquivalent = binaryOperatorEquivalent;
         }
 
         public String asString() {
             return codeRepresentation;
         }
 
         public Optional<BinaryExpr.Operator> toBinaryOperator() {
-            return Optional.ofNullable(binaryOperatorEquivalent);
+            switch (this) {
```

Comment:
what about searching over BinaryExpr.values for an element with a matching name?

Reply:
- less type safe (http://wiki.c2.com/?StringlyTyped)
- less dull code
- less likely to break when new operators are introduced
- less efficient

Another alternative: consulting a `Map<AssignExpr.Operator, Optional<BinaryExpr.Operator>>` in that method.

I have a light preference for the current solution because I consider it clean :-)

Reply:
I am happy with your solution, not feeling strongly about this

Reply:
Good, then I don't have to change it again ;-)